### ISSUE TITLE
Closes immediately after opening on Linux (Ubuntu MATE 22.04)

### ISSUE THREAD

**Issue by electricbrass** [nesbox/TIC-80#2001]
I installed the PRO deb and when I try to launch TIC-80, it shows up for a couple seconds and then immediately disappears (crashes?). If there's somewhere to look for a log or anything let me know.

I have tried both the PRO version of 1.0.2164, and 1.0.2241-dev.

Exit code is 141

**Comment by nesbox** [comment]
Try to call `tic80 --soft` to disable GPU rendering and see what happens.

**Comment by electricbrass** [comment]
Same result. Closes after a second or two, exit code 141.

**Comment by TomTheDragon** [comment]
I do have the same problem on Mint 21 Cinnamon, also full screen seems to be not working as well.
The free flatpak version however seems to work fine.

**Comment by adamlincoln** [comment]
I have the same problem in a source-built Pro version (on ref 7f4ad780d75d2cd8446f856f85ba293af70530eb).  Debian Linux (in a chrome OS linux container).  I took a few minutes to narrow down where it might be happening, and I can confirm that the callback at https://github.com/libuv/libuv/blob/9c3d692b3941a2a4171629fb52af2e1029c415e8/src/unix/linux-core.c#L462 gets called but never returns.

I will keep digging but I thought this info might help a maintainer understand what's happening.

**Comment by adamlincoln** [comment]
At the moment it looks like studio/net.c's onConnect() function gets called during this callback chain.  If we keep following we can see that eventually, deep inside libuv, this calls write() at line 717 of libuv/src/unix/stream.c, which never returns.  Based on my reading of write()'s manpage, this means that write() is setting EPIPE and raising SIGPIPE, consistent with exit code 141.

Now I'm looking for what should be on the other end of that pipe and why it's not listening.

**Comment by adamlincoln** [comment]
Instead of looking more at the other end of the pipe, I discovered [this conversation in libuv's repo](https://github.com/joyent/libuv/issues/1254) and then [two](https://github.com/HaxeFoundation/hashlink/pull/382) [examples](https://github.com/canonical/dqlite/pull/343) of projects that consume libuv setting SIG_IGN for SIGPIPE.  There's also [this warning](https://nikhilm.github.io/uvbook/filesystem.html#id3) in nikhilm's uvbook.  The problem only affects libuv in Linux, and the remedy appears to be to set `signal(SIGPIPE, SIG_IGN)`.

I've confirmed in a local pro build on Linux that `signal(SIGPIPE, SIG_IGN)` at the top of main.c's `start()` method eliminates the crash.  I can also confirm that the "save raw code to file" pro feature works fine, and I have no reason to think ignoring SIGPIPE causes any other problems with TIC-80.  But I'll leave it to @nesbox to decide whether this is the best plan for TIC-80, and the change is trivial so it is probably not worth it for me to open a pull request.

I hope this helps anyone who is seeing this crash on Linux.
